wayland/window.c

```diff
@@ -41,13 +41,23 @@ static void pointer_handle_button(void *data, struct wl_pointer *pointer, uint32
 	struct window *window = data;
 	struct pointer_input *input = &window->pointer_input;
 
-	if (window->pointer_input.notify) {
-		window->pointer_input.notify(window, input->last_x, input->last_y, button);
+	if (window->pointer_input.notify_button) {
+		window->pointer_input.notify_button(window, input->last_x, input->last_y, button);
 	}
 }
 
 static void pointer_handle_axis(void *data, struct wl_pointer *pointer,
-		    uint32_t time, uint32_t axis, wl_fixed_t value) {
+				uint32_t time, uint32_t axis, wl_fixed_t value) {
+	struct window *window = data;
+
+	if (!sway_assert(!axis, "Only support horizontal axis")) {
```

Comment:
Assertions are to detect things that should never happen. Horizontal scrolling doesn't qualify.